exporter/collector/internal/integrationtest/metric_test_server.go

```diff
@@ -78,23 +84,23 @@ func (m *MetricsTestServer) Serve() error {
 
 type fakeMetricServiceServer struct {
 	monitoringpb.UnimplementedMetricServiceServer
-	createMetricDescriptorChan chan<- *monitoringpb.CreateMetricDescriptorRequest
-	createTimeSeriesChan       chan<- *monitoringpb.CreateTimeSeriesRequest
+	appendCreateMetricDescriptorReq func(*monitoringpb.CreateMetricDescriptorRequest)
+	appendCreateTimeSeriesReq       func(*monitoringpb.CreateTimeSeriesRequest)
 }
 
 func (f *fakeMetricServiceServer) CreateTimeSeries(
 	ctx context.Context,
 	req *monitoringpb.CreateTimeSeriesRequest,
 ) (*emptypb.Empty, error) {
-	go func() { f.createTimeSeriesChan <- req }()
```

Comment:
I think increasing the channel buffer size and making this run in the same goroutine would also fix the issue, but idk how big the buffer may need to be. If it's too small, this solution would deadlock

Reply:
Ah yes sorry, the slice append should be in the same goroutine, let me update.

If I leave it as a channel, it can be in the same goroutine but AFAIK it will block if the buffer is full and cause deadlock

Reply:
Yeah, I think this works much better.